### PR TITLE
sysdump: Detect features from cilium-config ConfigMap

### PR DESCRIPTION
**Pull request by michi-covalent**
- Call FeatureSet.ExtractFromConfigMap to extract features from cilium-config.
- Add tasks related to SPIRE server if and only if "mutual-auth-spiffe" feature is enabled.

Ref: #1962

### CODE REVIEW

**Comment by michi-covalent** [comment]
requesting review from service mesh team 🚀🙏

File: sysdump/sysdump.go [hunk 1]
```diff
@@ -263,6 +267,14 @@ func NewCollector(k KubernetesClient, o Options, startTime time.Time, cliVersion
 			}
 			c.log("ℹ️  %s ConfigMap not found in %s namespace", ciliumConfigMapName, c.Options.CiliumNamespace)
 		}
+		if c.CiliumConfigMap != nil && len(c.CiliumPods) > 0 {
+			ciliumVersion, err := c.Client.GetCiliumVersion(context.Background(), c.CiliumPods[0])
```
Comment (derailed) [inline sysdump/sysdump.go:271]: Yikes! hopefully we'll have a better way to retrieve version info... perhaps helm values??

**Review by derailed — APPROVED** [review]
@michi-covalent Nice work!

File: sysdump/sysdump.go (hunk 1 above)
Comment (michi-covalent) [reply sysdump/sysdump.go:271]: yeah this is not great, but it needs to work even when you install cilium without doing `helm install`, like `helm template | kubectl apply -` or similar.

**Review by meyskens — APPROVED** [review]
LGTM from service mesh

**Pull request merged**